reduce_cluster_test.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2016 ScyllaDB
+
+import logging
+
+from avocado import main
+
+from sdcm.tester import ClusterTester
+from sdcm.tester import clean_aws_resources
+from sdcm.nemesis import DecommissionNoAddMonkey
+import time
+
+
+class ReduceClusterTest(ClusterTester):
+    """
+    Test scylla cluster reduction (removing nodes to shrink cluster from initial cluster size to target cluster size).
+
+    :avocado: enable
+    """
+
+    @clean_aws_resources
+    def setUp(self):
+        self.credentials = None
+        self.db_cluster = None
+        self.loaders = None
+        logging.getLogger('botocore').setLevel(logging.CRITICAL)
+        logging.getLogger('boto3').setLevel(logging.CRITICAL)
+        self.stress_thread = None
+
+    def wait_for_init(self, cluster_starting_size, cluster_target_size):
+        # We're starting the cluster with more than 3 nodes due to
+        # replication factor settings we're using with cassandra-stress
+        self._cluster_starting_size = cluster_starting_size
+        self._cluster_target_size = cluster_target_size
+        self.init_resources(n_db_nodes=self._cluster_starting_size, n_loader_nodes=1)
+        self.loaders.wait_for_init()
+        self.db_cluster.wait_for_init()
+
+    def reduce_cluster(self, cluster_starting_size, cluster_target_size=3):
+        self.wait_for_init(cluster_starting_size, cluster_target_size)
+        nodes_to_remove = self._cluster_starting_size - self._cluster_target_size
+        # FIXME: we should find a way to stop c-s when decommission is finished instead of estimating time to run
```

Comment:
Maybe we could send a SIGTERM to c-s. I still haven't seen what is the behavior in that case. If SIGTERM prompts c-s to end gracefully, that would be our solution.